packages/lit-dev-content/site/blog/2022-02-07-eleventy.md

```diff
@@ -0,0 +1,181 @@
+layout: blog-post.html
+title: "Eleventy + Lit"
+summary: "Announcing a new Lit Labs Eleventy plugin for static pre-rendering of Lit components"
+date: 2022-02-07
+author:
+  - al-marks
+---
+
+The Lit team is pleased to announce the first release of
+[`@lit-labs/eleventy-plugin-lit`](https://github.com/lit/lit/tree/main/packages/labs/eleventy-plugin-lit),
+a new plugin for [Eleventy](https://www.11ty.dev/) that pre-renders your Lit
+components as static HTML during your Eleventy build, and lets you re-hydrate
+them after your JavaScript loads.
+
+<center>
+
+![@lit-labs/eleventy-plugin-lit GitHub repo](/images/blog/eleventy/repo-screenshot.png){height="600" align="center"}
+
+</center>
+
+<aside class="warning">
+
+As with all Lit Labs projects, this release should be considered experimental
+and incomplete. We'd love for you to try it out and leave feedback, but expect
+to find a few rough edges.
+
+Check out the <a
+href="https://github.com/lit/lit/tree/main/packages/labs/eleventy-plugin-lit">roadmap</a>
+for a list of issues and missing features we're still working on.
+
+</aside>
+
+Check it out now on
+[GitHub](https://github.com/lit/lit/tree/main/packages/labs/eleventy-plugin-lit)
+and [NPM](https://www.npmjs.com/package/@lit-labs/eleventy-plugin-lit), or read
+on to learn more about what it does, how it works, and why you might want to use
+it.
+
+## What does it do?
+
+After you've added `@lit-labs/eleventy-plugin-lit` to your Eleventy config, then
+whenever you write a custom element tag in a markdown or HTML file, the initial
+state of that component's shadow DOM and styles will be rendered directly into
+the static HTML.
+
+For example, given a markdown file `hello.md`:
+
+```html
+# Greetings
+
+<demo-greeter name="World"></demo-greeter>
+```
+
+And a component definition `js/demo-greeter.js`:
+
+```js
+import {LitElement, html, css} from 'lit';
+
+class DemoGreeter extends LitElement {
+  static styles = css`
+    b { color: red; }
+  `;
+
+  static properties = {
+    name: {},
+  };
+
+  render() {
+    return html`Hello <b>${this.name}</b>!`;
+  }
+}
+customElements.define('demo-greeter', DemoGreeter);
+```
+
+Then Eleventy will produce an HTML file `greeting/index.html` like this:
+
+```html
+<h1>Greetings</h1>
+
+<demo-greeter name="World">
+  <template shadowroot="open">
+    <style>
+      b { color: red; }
+    </style>
+    Hello <b>World</b>!
+  </template>
+</demo-greeter>
+```
+
+This HTML allows the browser to paint the initial state of the component's DOM
+and styles, preserving all of the encapsulation boundaries of web components,
+even before a single line of JavaScript has downloaded!
+
+Later, when the JavaScript definition of `<demo-greeter>` has finally loaded,
+the pre-rendered component is upgraded to its fully interactive JavaScript
+implementation through a process called re-hydration.
+
+## Why should I use it?
+
+Pre-rendering components is an effective method for reducing the time it takes
```

Comment:
Maybe "can be an effective method"? We've definitely seen cases on large sites where deep SSR of components is _slower_ because of the additional HTML size.